File: myExpenses/src/main/java/org/totschnig/myexpenses/model/CurrencyContext.java
package org.totschnig.myexpenses.model;

import androidx.annotation.NonNull;

public interface CurrencyContext {
  @NonNull CurrencyUnit get(@NonNull String currencyCode);

  void storeCustomFractionDigits(String currencyCode, int fractionDigits);

  void storeCustomSymbol(String currencyCode, String symbol);

  void ensureFractionDigitsAreCached(CurrencyUnit currency);

  void invalidateHomeCurrency();
}

File: myExpenses/src/main/java/org/totschnig/myexpenses/model/CurrencyContext.kt
package org.totschnig.myexpenses.model

import androidx.annotation.VisibleForTesting
import kotlinx.coroutines.flow.Flow
import java.util.Currency

interface CurrencyContext {
    operator fun get(currencyCode: String): CurrencyUnit
    fun getAll(): Flow<List<CurrencyUnit>>
    fun invalidateHomeCurrency()
    fun invalidate(currencyCode: String)

    val homeCurrencyString: String
    val homeCurrencyUnit: CurrencyUnit
        get() = this[homeCurrencyString]
    val localCurrency: Currency
}
